akka-stream/src/main/scala/akka/stream/StreamTcpException.scala

```diff
@@ -7,8 +7,9 @@ import scala.util.control.NoStackTrace
 
 class StreamTcpException(msg: String) extends RuntimeException(msg) with NoStackTrace
 
-abstract class BindFailedException extends StreamTcpException("bind failed")
+class BindFailedException extends StreamTcpException("bind failed")
 
+@deprecated("BindFailedException object will never be thrown. Match on the class instead.")
```

Comment:
All those exception classes should have never been exposed as concrete classes in the first place.